src/adapter/src/coord.rs

```diff
@@ -1506,14 +1506,29 @@ impl Coordinator {
                 entry.item().typ(),
                 entry.id()
             );
-            let policy = entry.item().initial_logical_compaction_window();
+            let mut policy = entry.item().initial_logical_compaction_window();
             match entry.item() {
                 // Currently catalog item rebuild assumes that sinks and
                 // indexes are always built individually and does not store information
                 // about how it was built. If we start building multiple sinks and/or indexes
                 // using a single dataflow, we have to make sure the rebuild process re-runs
                 // the same multiple-build dataflow.
-                CatalogItem::Source(_) => {
+                CatalogItem::Source(source) => {
+                    // Propagate source compaction windows to subsources if needed.
+                    if source.custom_logical_compaction_window.is_none() {
```

Comment:
What if the compaction window for a source `is_some` but does not equal that of the source? Should we not update the compaction window then?

Reply:
Talked in person: this is working as intended. Subsources inherit only if they do not have their own rh set.